shm/alcohol.cpp

```diff
@@ -0,0 +1,27 @@
+#include "alcohol.hpp"
+#include <iostream>
+
+Alcohol::Alcohol(const std::string& name, size_t amount, size_t basePrice, size_t percentage)
+        : Cargo(name, amount, basePrice), percentage_(percentage) {} ;
+
+size_t Alcohol::getPrice() const {
+    return static_cast<size_t>(basePrice_ * static_cast<float>(percentage_)/static_cast<float>(MaxPercentage));
+}
+
+Cargo& Alcohol::operator+=(size_t amount) {
+    amount_ += amount;
+    return *this;
+}
```

Comment:
What happend when you add to much cargo (more then capacity?) Do you handle this somewhere? If yes please move also below logic to this function (with substract) because now you have logic for substract in Cargo and for add in other class this is not god :)